api/src/main/java/org/eclipse/microprofile/config/inject/ConfigProperties.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.config;
+
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Stereotype;
+
+
+/**
+ * <p>
+ * Retrieve a number of related configuration properties to a property interface.
+ * <pre>
+ * &#064;ConfigProperties("server")
+ * public interface MyServer {
+ *      String host; //maps the property name server.host
+ *      int port;    //maps to the property name server.port
+ *      String context; //maps to the property name server.context
+ *      &#064;ConfigProperty(name="oldServer.location")
+ *      String location; //maps to the property name oldServer.location
+ * }
+ * </pre>
+ *
+ * 
+ *
+ * @since 2.0
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
+ * 
+ */
+
+@Target({TYPE})
+@Retention(RUNTIME)
+@Documented
+@Stereotype
+@Dependent
+public @interface ConfigProperties {
+
+    /**
+     * Specify the configuration property prefix
+     * @return the configuration property prefix
+     */
+    String value() default "";
```

Comment:
Maybe call this `prefix`?

Reply:
If we leave use `value()`, the usage will be 

`@ConfigProperties("customer")`

If we change to `prefix`, the usage will be
`@ConfigProperties(prefix="customer")`.
I think the former is neater.

Reply:
It was in a perspective of extending it in the future with other fields.

Reply:
I have updated accordingly.